nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumerPool.java

```diff
@@ -202,20 +196,17 @@ public ConsumerPool(
                             readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig);
                             readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
                         }
-                    } catch (Exception e) {
-                        readerGroupManager.close();
-                        throw e;
                     }
                 } catch (Exception e) {
-                    initiateCheckpointExecutor.shutdown();
+                    readerGroupManager.close();
                     throw e;
                 }
             } catch (Exception e) {
-                performCheckpointExecutor.shutdown();
+                initiateCheckpointExecutor.shutdown();
                 throw e;
             }
         } catch (Exception e) {
-            clientFactory.close();
```

Comment:
`ClientFactory` was no longer needed, so the outer `try` level was able to be removed.